### ISSUE TITLE
Criteria to sort pictures/videos

### ISSUE THREAD

**Issue by darguez** [nextcloud/photos#1424]
I know this topic is not new, but it is still not clear to me what is the criteria for sorting the photos / videos.

The following are real situations with different criteria:

1.  Photos/videos are always taken with devices that store the files as YYYYMMDD-HHMMSS.jpg --> Filename is goot to sort
2. Photos/videos are stored in nextcloud as they are taken --> Creation date is good to sort
3. Photos/videos are stored in nextcloud when wifi is available, some uploads fails and need retries, photos/videos are not stored in nextcloud in the order they were taken. --> Date taken/recorded is good to sort
4. Photos/videos are uploaded manually not in the order they were taken --> Date taken/recorded is good to sort

I know sort by date taken/recorded implies to manage exif data and maybe this option is not available. But in those cases the photos are not correctly displayed by default, the user should have some alternative:
- Rename files to assure they are in the correct order
- Change file date to allow sorting by desired date. The problem here is that generally it is not possible to change the creation date but the modified date. And if the app only allows sorting by creation date there is nothing the user can do to fix the order.

In conclusion, there are different scenarios that require different criteria, but it seems at least one alternative (sort by name or sort by modified date) should be available to fix sorting by creation date when it doesn't provide the expected results.

Finally, just to mention that the order from newest to oldest instead oldest to newest has a negative impact in slideshow, because you are forced to see the photos in reserve order.

**Comment by darguez** [comment]
BTW I discovered that after executing this script: https://github.com/darguez/media-sorting and re-scan the corresponding directories, I can get the Album sorted from oldest to newest by adding the line:
`->orderBy('mtime','ASC')` in ` getForAlbumIdAndUserWithFiles()` function in `photos/lib/Album/AlbumMapper.php` file

**Comment by jtrees** [comment]
> BTW I discovered that after executing this script: https://github.com/darguez/media-sorting and re-scan the corresponding directories, I can get the Album sorted from oldest to newest by adding the line: `->orderBy('mtime','ASC')` in ` getForAlbumIdAndUserWithFiles()` function in `photos/lib/Album/AlbumMapper.php` file

This is a horrible workaround... but incredibly useful. Thanks for sharing!

**Comment by paulvt** [comment]
Useful indeed! It unfortunately does not seem to work when using the public link to view the albums, so the sorting is not applied everywhere but could not find where it is missing.

**Comment by darguez** [comment]
Other functions where you can add `->orderBy('mtime','ASC')`:
- `getForAlbumIdAndUserWithFiles()` <-- the one mentioned above
- `getForAlbumIdAndFileId()`
- `getSharedAlbumsForCollaboratorWithFiles()`

**Comment by joshtrichards** [comment]
Hi @darguez - This topic is already being tracked in #87 and #397. Please close out this Issue to avoid confusion / duplicate work (unless you have a good reason to keep it separate that I'm overlooking).

**Comment by darguez** [comment]
ok, thanks for contributing to solve this issue!
